File: shared/main/java/net/ccbluex/liquidbounce/ui/client/GuiAntiModDisable.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.ui.client;

import java.io.IOException;
import java.util.Collection;

import net.ccbluex.liquidbounce.LiquidBounce;
import net.ccbluex.liquidbounce.api.minecraft.client.gui.IGuiButton;
import net.ccbluex.liquidbounce.api.minecraft.client.gui.IGuiScreen;
import net.ccbluex.liquidbounce.api.util.WrappedGuiScreen;
import net.ccbluex.liquidbounce.features.special.AntiModDisable;
import net.ccbluex.liquidbounce.file.FileManager;
import net.ccbluex.liquidbounce.ui.font.Fonts;

import org.lwjgl.input.Keyboard;

public class GuiAntiModDisable extends WrappedGuiScreen
{
	public static final String BLOCK_FML = "Block FML ";
	public static final String BLOCK_FMLPROXY_PACKET = "Block FMLProxyPackets ";
	public static final String SPOOF_MC_BRAND_PAYLOAD_PACKETS = "Spoof FML ClientBrandRetriever Payloads ";
	public static final String BLOCK_WORLD_DOWNLOADER_PAYLOAD_PACKETS = "Block WorldDownloader Payloads ";
	public static final String BLOCK_THE_5_ZIG_S_MOD_PAYLOAD_PACKETS = "Block The 5zig's mod Payloads ";
	public static final String BLOCK_BETTER_SPRINTING_PAYLOAD_PACKETS = "Block Better Sprinting mod Payloads ";
	public static final String BLOCK_CRACKED_VAPE_SABOTAGES = "Block Cracked Vape Sabotages ";
	public static final String BLOCK_DIPERMISSIONS_PAYLOAD_PACKETS = "Block DIPermissions Payloads ";
	public static final String BLOCK_PERMISSIONS_REPL_PAYLOAD_PACKETS = "Block PermissionsRepl Payloads ";
	public static final String BLOCK_SCHEMATICA_PAYLOAD_PACKETS = "Block Schematica Payloads ";
	public static final String PRINT_BLOCKED_SPOOFED_PACKETS = "Print Blocked/Spoofed Packets On The Chat ";

	private final IGuiScreen prevGui;

	private IGuiButton enabledButton;
	private IGuiButton fmlButton;
	private IGuiButton fmlProxyPacket;
	private IGuiButton clientBrandPayloadPacket;
	private IGuiButton wdlPayloadButton;
	private IGuiButton _5zigPayloadButton;
	private IGuiButton debugmode;
	private IGuiButton betterSprintingButton;
	private IGuiButton vapeButton;
	private IGuiButton dipermsButton;
	private IGuiButton permsreplButton;
	private IGuiButton schematicaButton;

	public GuiAntiModDisable(final IGuiScreen prevGui)
	{
		this.prevGui = prevGui;
	}

	@Override
	public void initGui()
	{
		final Collection<IGuiButton> buttonsList = representedScreen.getButtonList();
		final int width = representedScreen.getWidth();
		final int height = representedScreen.getHeight();

		buttonsList.add(enabledButton = classProvider.createGuiButton(1, width / 2 - 100, representedScreen.getHeight() / 4 + 35, "Enabled " + (AntiModDisable.enabled ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(fmlButton = classProvider.createGuiButton(2, width / 2 - 100, height / 4 + 50 + 25, BLOCK_FML + (AntiModDisable.enabled && AntiModDisable.blockFMLPackets ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(fmlProxyPacket = classProvider.createGuiButton(3, width / 2 - 100, height / 4 + 50 + 50, BLOCK_FMLPROXY_PACKET + (AntiModDisable.enabled && AntiModDisable.blockFMLProxyPackets ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(clientBrandPayloadPacket = classProvider.createGuiButton(4, width / 2 - 100, height / 4 + 50 + 75, SPOOF_MC_BRAND_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockClientBrandRetrieverPackets ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(wdlPayloadButton = classProvider.createGuiButton(5, width / 2 - 100, height / 4 + 50 + 100, BLOCK_WORLD_DOWNLOADER_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockWDLPayloads ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(_5zigPayloadButton = classProvider.createGuiButton(6, width / 2 - 100, height / 4 + 50 + 125, "Block The 5zig's mod Payload Packets " + (AntiModDisable.enabled && AntiModDisable.block5zigsmodPayloads ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(betterSprintingButton = classProvider.createGuiButton(7, width / 2 - 100, height / 4 + 50 + 150, BLOCK_BETTER_SPRINTING_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockBetterSprintingPayloads ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(vapeButton = classProvider.createGuiButton(8, width / 2 - 100, height / 4 + 50 + 175, BLOCK_CRACKED_VAPE_SABOTAGES + (AntiModDisable.enabled && AntiModDisable.blockCrackedVapeSabotages ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(dipermsButton = classProvider.createGuiButton(9, width / 2 - 100, height / 4 + 50 + 200, BLOCK_DIPERMISSIONS_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockDIPermissionsPayloads ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(permsreplButton = classProvider.createGuiButton(10, width / 2 - 100, height / 4 + 50 + 225, BLOCK_PERMISSIONS_REPL_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockPermissionsReplPayloads ? "\u00A7a(On)" : "\u00A7c(Off)")));
		buttonsList.add(schematicaButton = classProvider.createGuiButton(11, width / 2 - 100, height / 4 + 50 + 250, BLOCK_SCHEMATICA_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockSchematicaPayloads ? "\u00A7a(On)" : "\u00A7c(Off)")));

		buttonsList.add(debugmode = classProvider.createGuiButton(999, width / 2 - 100, height / 4 + 50 + 275, PRINT_BLOCKED_SPOOFED_PACKETS + (AntiModDisable.enabled && AntiModDisable.debug ? "\u00A7a(On)" : "\u00A7c(Off)")));

		buttonsList.add(classProvider.createGuiButton(0, width / 2 - 100, height / 4 + 55 + 300 + 5, "Back"));
	}

	@Override
	public void actionPerformed(final IGuiButton button)
	{
		switch (button.getId())
		{
			case 1:
				AntiModDisable.enabled = !AntiModDisable.enabled;
				enabledButton.setDisplayString("Enabled (" + (AntiModDisable.enabled ? "On" : "Off") + ")");
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				updateAll();
				break;
			case 2:
				AntiModDisable.blockFMLPackets = !AntiModDisable.blockFMLPackets;
				fmlButton.setDisplayString("Block FML (" + (AntiModDisable.blockFMLPackets ? "On" : "Off") + ")");
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 3:
				AntiModDisable.blockFMLProxyPackets = !AntiModDisable.blockFMLProxyPackets;
				fmlProxyPacket.setDisplayString("Block FMLProxyPackets (" + (AntiModDisable.blockFMLProxyPackets ? "On" : "Off") + ")");
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 4:
				AntiModDisable.blockClientBrandRetrieverPackets = !AntiModDisable.blockClientBrandRetrieverPackets;
				clientBrandPayloadPacket.setDisplayString("Block FML ClientBrandRetriever Packets (" + (AntiModDisable.blockClientBrandRetrieverPackets ? "On" : "Off") + ")");
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 5:
				AntiModDisable.blockWDLPayloads = !AntiModDisable.blockWDLPayloads;
				wdlPayloadButton.setDisplayString(BLOCK_WORLD_DOWNLOADER_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockWDLPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 6:
				AntiModDisable.block5zigsmodPayloads = !AntiModDisable.block5zigsmodPayloads;
				_5zigPayloadButton.setDisplayString(BLOCK_THE_5_ZIG_S_MOD_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.block5zigsmodPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 7:
				AntiModDisable.blockBetterSprintingPayloads = !AntiModDisable.blockBetterSprintingPayloads;
				betterSprintingButton.setDisplayString(BLOCK_BETTER_SPRINTING_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockBetterSprintingPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 8:
				AntiModDisable.blockCrackedVapeSabotages = !AntiModDisable.blockCrackedVapeSabotages;
				vapeButton.setDisplayString(BLOCK_CRACKED_VAPE_SABOTAGES + (AntiModDisable.enabled && AntiModDisable.blockCrackedVapeSabotages ? "\u00A7a(On)" : "\u00A7c(Off)"));
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 9:
				AntiModDisable.blockDIPermissionsPayloads = !AntiModDisable.blockDIPermissionsPayloads;
				dipermsButton.setDisplayString(BLOCK_DIPERMISSIONS_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockDIPermissionsPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 10:
				AntiModDisable.blockPermissionsReplPayloads = !AntiModDisable.blockPermissionsReplPayloads;
				permsreplButton.setDisplayString(BLOCK_PERMISSIONS_REPL_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockPermissionsReplPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 11:
				AntiModDisable.blockSchematicaPayloads = !AntiModDisable.blockSchematicaPayloads;
				schematicaButton.setDisplayString(BLOCK_SCHEMATICA_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockSchematicaPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 999:
				AntiModDisable.debug = !AntiModDisable.debug;
				debugmode.setDisplayString(PRINT_BLOCKED_SPOOFED_PACKETS + (AntiModDisable.enabled && AntiModDisable.debug ? "\u00A7a(On)" : "\u00A7c(Off)"));
				FileManager.saveConfig(LiquidBounce.fileManager.valuesConfig);
				break;
			case 0:
				mc.displayGuiScreen(prevGui);
				break;
		}
	}

	private void updateAll()
	{
		fmlButton.setDisplayString(BLOCK_FML + (AntiModDisable.enabled && AntiModDisable.blockFMLPackets ? "\u00A7a(On)" : "\u00A7c(Off)"));
		fmlProxyPacket.setDisplayString(BLOCK_FMLPROXY_PACKET + (AntiModDisable.enabled && AntiModDisable.blockFMLProxyPackets ? "\u00A7a(On)" : "\u00A7c(Off)"));
		clientBrandPayloadPacket.setDisplayString(SPOOF_MC_BRAND_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockClientBrandRetrieverPackets ? "\u00A7a(On)" : "\u00A7c(Off)"));
		wdlPayloadButton.setDisplayString(BLOCK_WORLD_DOWNLOADER_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockWDLPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
		_5zigPayloadButton.setDisplayString(BLOCK_THE_5_ZIG_S_MOD_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.block5zigsmodPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
		betterSprintingButton.setDisplayString(BLOCK_BETTER_SPRINTING_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockBetterSprintingPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
		vapeButton.setDisplayString(BLOCK_CRACKED_VAPE_SABOTAGES + (AntiModDisable.enabled && AntiModDisable.blockCrackedVapeSabotages ? "\u00A7a(On)" : "\u00A7c(Off)"));
		dipermsButton.setDisplayString(BLOCK_DIPERMISSIONS_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockDIPermissionsPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
		permsreplButton.setDisplayString(BLOCK_PERMISSIONS_REPL_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockPermissionsReplPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
		schematicaButton.setDisplayString(BLOCK_SCHEMATICA_PAYLOAD_PACKETS + (AntiModDisable.enabled && AntiModDisable.blockSchematicaPayloads ? "\u00A7a(On)" : "\u00A7c(Off)"));
		debugmode.setDisplayString(PRINT_BLOCKED_SPOOFED_PACKETS + (AntiModDisable.enabled && AntiModDisable.debug ? "\u00A7a(On)" : "\u00A7c(Off)"));
	}

	@Override
	public void drawScreen(final int mouseX, final int mouseY, final float partialTicks)
	{
		representedScreen.drawBackground(0);
		Fonts.fontBold180.drawCenteredString("AntiModDisable", (int) (representedScreen.getWidth() / 2.0F), (int) (representedScreen.getHeight() / 10.0F + 5.0F), 4673984, true);

		super.drawScreen(mouseX, mouseY, partialTicks);
	}

	@Override
	public void keyTyped(final char typedChar, final int keyCode) throws IOException
	{
		if (keyCode == Keyboard.KEY_ESCAPE)
		{
			mc.displayGuiScreen(prevGui);
			return;
		}

		super.keyTyped(typedChar, keyCode);
	}
}

File: shared/main/java/net/ccbluex/liquidbounce/ui/client/GuiAntiModDisable.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.ui.client

import net.ccbluex.liquidbounce.LiquidBounce
import net.ccbluex.liquidbounce.api.minecraft.client.gui.IGuiButton
import net.ccbluex.liquidbounce.api.minecraft.client.gui.IGuiScreen
import net.ccbluex.liquidbounce.api.util.WrappedGuiScreen
import net.ccbluex.liquidbounce.features.special.AntiModDisable
import net.ccbluex.liquidbounce.file.FileManager.Companion.saveConfig
import net.ccbluex.liquidbounce.ui.font.Fonts
import org.lwjgl.input.Keyboard
import java.io.IOException

class GuiAntiModDisable(private val prevGui: IGuiScreen) : WrappedGuiScreen()
{
	private lateinit var enabledButton: IGuiButton
	private lateinit var fmlButton: IGuiButton
	private lateinit var fmlProxyPacket: IGuiButton
	private lateinit var clientBrandPayloadPacket: IGuiButton
	private lateinit var wdlPayloadButton: IGuiButton
	private lateinit var `5zigPayloadButton`: IGuiButton
	private lateinit var debugmode: IGuiButton
	private lateinit var betterSprintingButton: IGuiButton
	private lateinit var vapeButton: IGuiButton
	private lateinit var dipermsButton: IGuiButton
	private lateinit var permsreplButton: IGuiButton
	private lateinit var schematicaButton: IGuiButton
	override fun initGui()
	{
		val buttonsList: MutableCollection<IGuiButton> = representedScreen.buttonList
		val width = representedScreen.width
		val height = representedScreen.height

		val buttonX = (width shr 1) - 100
		val buttonY = (height shr 2) + 50

		buttonsList.add(classProvider.createGuiButton(1, buttonX, (representedScreen.height shr 2) + 35, "Enabled ${getStateString(AntiModDisable.enabled)}").also { enabledButton = it })

		val addButton = { id: Int, index: Int, text: String, callback: (IGuiButton) -> Unit ->
			buttonsList.add(classProvider.createGuiButton(id, buttonX, buttonY + 25 * index + 25, text).also(callback))
		}

		addButton(2, 0, "$BLOCK_FML${getStateString(AntiModDisable.enabled && AntiModDisable.blockFMLPackets)}") { fmlButton = it }
		addButton(3, 1, "$BLOCK_FML_PROXY_PACKET${getStateString(AntiModDisable.enabled && AntiModDisable.blockFMLProxyPackets)}") { fmlProxyPacket = it }
		addButton(4, 2, "$SPOOF_BRAND_PAYLOAD_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockClientBrandRetrieverPackets)}") { clientBrandPayloadPacket = it }
		addButton(5, 3, "$BLOCK_WDL_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockWDLPayloads)}") { wdlPayloadButton = it }
		addButton(6, 4, "$BLOCK_5ZIG_PACKETS ${getStateString(AntiModDisable.enabled && AntiModDisable.block5zigsmodPayloads)}") { `5zigPayloadButton` = it }
		addButton(7, 5, "$BLOCK_BETTERSPRINTING_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockBetterSprintingPayloads)}") { betterSprintingButton = it }
		addButton(8, 6, "$BLOCK_VAPE_SABOTAGES${getStateString(AntiModDisable.enabled && AntiModDisable.blockCrackedVapeSabotages)}") { vapeButton = it }
		addButton(9, 7, "$BLOCK_DIPERMISSIONS_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockDIPermissionsPayloads)}") { dipermsButton = it }
		addButton(10, 8, "$BLOCK_PERMISSIONS_REPL_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockReplicatedPermissionsPayloads)}") { permsreplButton = it }
		addButton(11, 9, "$BLOCK_SCHEMATICA_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockSchematicaPayloads)}") { schematicaButton = it }
		addButton(999, 10, "$PRINT_DEBUG_MESSAGES${getStateString(AntiModDisable.enabled && AntiModDisable.debug)}") { debugmode = it }

		buttonsList.add(classProvider.createGuiButton(0, buttonX, (height shr 2) + 55 + 300 + 5, "Back"))
	}

	override fun actionPerformed(button: IGuiButton)
	{
		val buttonID = button.id

		when (buttonID)
		{
			1 ->
			{
				AntiModDisable.enabled = !AntiModDisable.enabled
				enabledButton.displayString = "Enabled ${getStateString(AntiModDisable.enabled)}"
				updateAll()
			}

			2 ->
			{
				AntiModDisable.blockFMLPackets = !AntiModDisable.blockFMLPackets
				fmlButton.displayString = "$BLOCK_FML${getStateString(AntiModDisable.blockFMLPackets)}"
			}

			3 ->
			{
				AntiModDisable.blockFMLProxyPackets = !AntiModDisable.blockFMLProxyPackets
				fmlProxyPacket.displayString = "$BLOCK_FML_PROXY_PACKET${getStateString(AntiModDisable.blockFMLProxyPackets)}"
			}

			4 ->
			{
				AntiModDisable.blockClientBrandRetrieverPackets = !AntiModDisable.blockClientBrandRetrieverPackets
				clientBrandPayloadPacket.displayString = "$SPOOF_BRAND_PAYLOAD_PACKETS${getStateString(AntiModDisable.blockClientBrandRetrieverPackets)}"
			}

			5 ->
			{
				AntiModDisable.blockWDLPayloads = !AntiModDisable.blockWDLPayloads
				wdlPayloadButton.displayString = "$BLOCK_WDL_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockWDLPayloads)}"
			}

			6 ->
			{
				AntiModDisable.block5zigsmodPayloads = !AntiModDisable.block5zigsmodPayloads
				`5zigPayloadButton`.displayString = "$BLOCK_5ZIG_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.block5zigsmodPayloads)}"
			}

			7 ->
			{
				AntiModDisable.blockBetterSprintingPayloads = !AntiModDisable.blockBetterSprintingPayloads
				betterSprintingButton.displayString = "$BLOCK_BETTERSPRINTING_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockBetterSprintingPayloads)}"
			}

			8 ->
			{
				AntiModDisable.blockCrackedVapeSabotages = !AntiModDisable.blockCrackedVapeSabotages
				vapeButton.displayString = "$BLOCK_VAPE_SABOTAGES${getStateString(AntiModDisable.enabled && AntiModDisable.blockCrackedVapeSabotages)}"
			}

			9 ->
			{
				AntiModDisable.blockDIPermissionsPayloads = !AntiModDisable.blockDIPermissionsPayloads
				dipermsButton.displayString = "$BLOCK_DIPERMISSIONS_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockDIPermissionsPayloads)}"
			}

			10 ->
			{
				AntiModDisable.blockReplicatedPermissionsPayloads = !AntiModDisable.blockReplicatedPermissionsPayloads
				permsreplButton.displayString = "$BLOCK_PERMISSIONS_REPL_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockReplicatedPermissionsPayloads)}"
			}

			11 ->
			{
				AntiModDisable.blockSchematicaPayloads = !AntiModDisable.blockSchematicaPayloads
				schematicaButton.displayString = "$BLOCK_SCHEMATICA_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockSchematicaPayloads)}"
			}

			999 ->
			{
				AntiModDisable.debug = !AntiModDisable.debug
				debugmode.displayString = "$PRINT_DEBUG_MESSAGES${getStateString(AntiModDisable.enabled && AntiModDisable.debug)}"
			}

			0 -> mc.displayGuiScreen(prevGui)
		}

		if (buttonID != 0) saveConfig(LiquidBounce.fileManager.valuesConfig) // Save the changed value
	}

	private fun updateAll()
	{
		fmlButton.displayString = "$BLOCK_FML${getStateString(AntiModDisable.enabled && AntiModDisable.blockFMLPackets)}"
		fmlProxyPacket.displayString = "$BLOCK_FML_PROXY_PACKET${getStateString(AntiModDisable.enabled && AntiModDisable.blockFMLProxyPackets)}"
		clientBrandPayloadPacket.displayString = "$SPOOF_BRAND_PAYLOAD_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockClientBrandRetrieverPackets)}"
		wdlPayloadButton.displayString = "$BLOCK_WDL_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockWDLPayloads)}"
		`5zigPayloadButton`.displayString = "$BLOCK_5ZIG_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.block5zigsmodPayloads)}"
		betterSprintingButton.displayString = "$BLOCK_BETTERSPRINTING_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockBetterSprintingPayloads)}"
		vapeButton.displayString = "$BLOCK_VAPE_SABOTAGES${getStateString(AntiModDisable.enabled && AntiModDisable.blockCrackedVapeSabotages)}"
		dipermsButton.displayString = "$BLOCK_DIPERMISSIONS_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockDIPermissionsPayloads)}"
		permsreplButton.displayString = "$BLOCK_PERMISSIONS_REPL_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockReplicatedPermissionsPayloads)}"
		schematicaButton.displayString = "$BLOCK_SCHEMATICA_PACKETS${getStateString(AntiModDisable.enabled && AntiModDisable.blockSchematicaPayloads)}"
		debugmode.displayString = "$PRINT_DEBUG_MESSAGES${getStateString(AntiModDisable.enabled && AntiModDisable.debug)}"
	}

	override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float)
	{
		representedScreen.drawBackground(0)
		Fonts.fontBold180.drawCenteredString("AntiModDisable", (representedScreen.width shr 1).toFloat(), representedScreen.height * 0.1f + 5.0f, 4673984, true)

		super.drawScreen(mouseX, mouseY, partialTicks)
	}

	@Throws(IOException::class)
	override fun keyTyped(typedChar: Char, keyCode: Int)
	{
		if (keyCode == Keyboard.KEY_ESCAPE)
		{
			mc.displayGuiScreen(prevGui)
			return
		}
		super.keyTyped(typedChar, keyCode)
	}

	companion object
	{
		// Button string constants
		const val BLOCK_FML = "Block FML "
		const val BLOCK_FML_PROXY_PACKET = "Block FMLProxyPackets "
		const val SPOOF_BRAND_PAYLOAD_PACKETS = "Spoof FML ClientBrandRetriever Payloads "
		const val BLOCK_WDL_PACKETS = "Block WorldDownloader Payloads "
		const val BLOCK_5ZIG_PACKETS = "Block The 5zig's mod Payloads "
		const val BLOCK_BETTERSPRINTING_PACKETS = "Block Better Sprinting mod Payloads "
		const val BLOCK_VAPE_SABOTAGES = "Block Cracked Vape Sabotages "
		const val BLOCK_DIPERMISSIONS_PACKETS = "Block DIPermissions Payloads "
		const val BLOCK_PERMISSIONS_REPL_PACKETS = "Block PermissionsRepl Payloads "
		const val BLOCK_SCHEMATICA_PACKETS = "Block Schematica Payloads "
		const val PRINT_DEBUG_MESSAGES = "Print Blocked/Spoofed Packets On The Chat "

		@JvmStatic
		private fun getStateString(state: Boolean): String = if (state) "\u00A7a(On)" else "\u00A7c(Off)"
	}
}
